packages/ui/src/Image.tsx

```diff
@@ -30,15 +51,27 @@ export const Image = forwardRef(function Image(
   );
 
   useEffect(() => {
-    setImageLoadFailed(false);
-  }, [props.src]);
+    console.log('useEffect triggered:', props.src);
+
+    const GlobalImage = window.Image;
+    const img = new GlobalImage();
+    img.src = props.src || placeHolderImage;
+    img.onload = handleLoad;
+    return () => {
+      img.onload = null;
+    };
+  }, [props.src, handleLoad]);
+
+  console.log('Rendering with states:', { src, isLoaded, imageLoadFailed });
 
   return (
     <img
       {...props}
-      src={
-        imageLoadFailed ? `${STATIC_IMAGES_URL}/placeholder.webp` : props.src
-      }
+      src={imageLoadFailed ? `${STATIC_IMAGES_URL}/placeholder.webp` : src}
+      style={{
+        filter: isLoaded ? 'none' : 'blur(20px)',
+        transition: isLoaded ? 'filter 0.5s ease-out' : 'none'
```

Comment:
we can remove transitions imo 🙇🏼

Reply:
this adds huge shadow like behavior 🙇🏼

Reply:
instead if we can reduce duration of transition, because if we completely remove there will be sudden change from blur to image , that not looks good

Reply:
0.1s ?